Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -117,6 +117,13 @@ def _root_.SMulHomClass.toMulActionHom [SMul M X] [SMul M Y] [SMulHomClass F M X
 instance [SMul M X] [SMul M Y] [SMulHomClass F M X Y] : CoeTC F (X →[M] Y) :=
   ⟨SMulHomClass.toMulActionHom⟩
 
+variable (M' X Y F) in
+/-- If Y/X/M' forms a scalar tower, any map X → Y preserving X-action also preserves M-action. -/
+def _root_.IsScalarTower.smulHomClass [MulOneClass X] [SMul X Y] [IsScalarTower M' X Y]
+    [SMulHomClass F X X Y] : SMulHomClass F M' X Y where
+  map_smul f m x := by
+    rw [← mul_one (m • x), ← smul_eq_mul, map_smul, smul_assoc, ← map_smul, smul_eq_mul, mul_one]
```

Comment:
Not sure why this can't be an instance?

Reply:
I believe it is because they are outparams?

Reply:
Okay I think I get how it works: one F is supposed to correspond to one M' only. Not very flexible! X and Y certainly should be outParams but maybe M' could be an autoParam instead?

Reply:
Does autoParam work in instances? Anyway I think such instances are probably out of scope of what `SMulHomClass` was trying to achieve.

Reply:
Thanks indeed the docs for [autoParam](https://leanprover-community.github.io/mathlib4_docs/Init/Meta.html#autoParam) says it won't affect typeclass resolution like optParam.